### PR TITLE
Use `ar` & `tar` instead of `dpkg -x`

### LINKED ISSUE #85: switch from `dpkg -x` to `ar x && tar -xvf`
installing a standalone build is useful for 

it is possible to extract without using `dpkg` and just using `ar` and `tar` which should be available on any of target systems (whenever `dpkg` - only on Debians):

```
(datalad) [d31548v@discovery7 8.20210803+git153-g63c2cfd47-1]$ ar x /dartfs-hpc/rc/home/v/d31548v/proj/datalad/ci/git-annex/builds/2021/08/cron-20210830/build-ubuntu.yaml-397-69466103-success/git-annex-debianstandalone-packages_8.20210803+git153-g63c2cfd47_amd64/git-annex-standalone_8.20210803+git153-g63c2cfd47-1~ndall+1_amd64.deb
(datalad) [d31548v@discovery7 8.20210803+git153-g63c2cfd47-1]$ ls
control.tar.gz  data.tar.gz  debian-binary
(datalad) [d31548v@discovery7 8.20210803+git153-g63c2cfd47-1]$ tar -xzf data.tar.gz 
(datalad) [d31548v@discovery7 8.20210803+git153-g63c2cfd47-1]$ ls
control.tar.gz  data.tar.gz  debian-binary  etc  usr
```

so I guess `ar x` should first extract into a tempdir and then `tar` could extract data.tar.gz into the target `install-dir` path, and then tempdir be cleaned up.

### PR DESCRIPTION
**Pull request by jwodder**
Closes #85.

### CODE REVIEW

File: src/datalad_installer.py [hunk 1]
```diff
@@ -2076,20 +2076,26 @@ def install_deb(
...
+        assert os.path.isabs(debpath)
         install_dir.mkdir(parents=True, exist_ok=True)
-        cmd.append("-x")
-        cmd.append(debpath)
-        cmd.append(install_dir)
-        runcmd(*cmd)
+        install_dir = install_dir.resolve()
+        with tempfile.TemporaryDirectory() as tmpdir:
+            oldpwd = os.getcwd()
+            os.chdir(tmpdir)
```
Comment (yarikoptic) [inline src/datalad_installer.py:2095]: note: although unlikely to matter here `chdir` within a process is a bit "atrocious" , e.g. IIRC `getcwd` would return a resolved directory name if you are in a dir which is symlinked somewhere.  I thought to suggest to use `--output` option for `ar` I find available on Debian  2.35.2, but it seems to be not available on centos with 2.27-27.base.el7.  So let's keep `chdir`ing I guess

**Comment by yarikoptic** [comment]
Looks LGTM, thank you!
just to make sure (since we have no test coverage for this) - did you try it out on some linux box?

**Comment by jwodder** [comment]
@yarikoptic Yes, I tested it in a Docker container.

**Comment by yarikoptic** [comment]
Great, thank you! Let's proceed

**Pull request merged**